misc/windows-deploy/hostsetup.ps1

```diff
@@ -11,41 +11,119 @@
+    } catch {
+        sleep 1
+    } finally {
+        $maxTries--
+    }
+}
 
-$dockerSubnet = (docker network inspect nat | ConvertFrom-Json).IPAM.Config.Subnet
+[int]$ifIndex = $netAdapter.ifIndex
+$dockerNatConfig = Invoke-Expression -Command "docker network inspect nat" | ConvertFrom-Json
+$dockerSubnet = $dockerNatConfig.IPAM.Config.Subnet
 
 # This address will only exist on systems that have already set up the routes.
-$ip = (Get-NetRoute -InterfaceIndex $ifIndex -DestinationPrefix $dockerSubnet)
-if(!($ip)) {
+$ip = (Get-NetRoute -InterfaceIndex $ifIndex -DestinationPrefix $dockerSubnet -ErrorAction:Ignore)
+# ip = $null when the route does not exist
 
-	# This command tells the APIPA interface that this IP exists.
-	New-NetIPAddress -InterfaceIndex $ifIndex -IPAddress $credentialAddress -PrefixLength 32
+if(!($ip)) { #TODO: Make each setup/config item idempotent rather than a simple assumption/check for the route
+    # This command creates a virtual ip for the APIPA interface
+    LogMsg "Configuring new virtual network adapter ip..."
+    New-NetIPAddress -InterfaceIndex $ifIndex -IPAddress $CREDENTIAL_ADDRESS -PrefixLength 32
+    LogMsg "Virtual network adapter ip created. Waiting for it to become available on the device..."
+    [int]$delay = 1
+    [int]$maxTries = 10
+    [bool]$available = $false
+    While (-not $available -and $maxTries -ge 0) {
+        try {
+            [string[]]$addresses = (Get-NetIPAddress -InterfaceIndex $ifIndex -AddressFamily:IPv4).IPAddress
+            if ($addresses.count -eq 1 -and $addresses -eq $CREDENTIAL_ADDRESS) {
+                LogMsg "Ip available: $((Get-NetIPAddress -InterfaceIndex $ifIndex -AddressFamily:IPv4).IPAddress)"
+                $available = $true
+            } else {
+                # Race condition where the adapter has multiple addresses before our new ip is assigned
+                # Note: Allowing this race condition can cause the netsh interface portproxy setup to
+                #       result in the network adapter in an unrecoverable bad state where the proxy
+                #       is unable to listen on port 80 because the port is in use by a non-functional
+                #       proxy rule.
+                sleep 1
+            }
+        } catch {
+            # wait for the adapter to become available
+            LogMsg "broke"
+            sleep 1
+        } finally {
+            $maxTries--
+        }
+    }
+    LogMsg "Virtual network adapter ip configuration complete"
 
-	# Enable the default docker IP range to be routable by the APIPA interface.
-	New-NetRoute -DestinationPrefix $dockerSubnet -ifIndex $ifindex
+    # Enable the default docker IP range to be routable by the APIPA interface.
+    LogMsg "Setting up new route for $($dockerSubnet) on adapter $($ifIndex)..."
+    New-NetRoute -DestinationPrefix $dockerSubnet -ifIndex $ifindex
+    # TODO: Validate via 'Get-NetRoute -DestinationPrefix $dockerSubnet -InterfaceIndex $ifIndex'
```

Comment:
Can this validation be added with this PR?

Reply:
Yes. I can work on that. Shouldnt take long.